File: 0283-move-zeroes/0283-move-zeroes.java
class Solution {
    public void moveZeroes(int[] arr) {
        int j = 0,i=0;
        while(i<arr.length){
            if(arr[i] != 0){
                if(i!=j){
                    int temp = arr[i];
                    arr[i] = arr[j];
                    arr[j] = temp;
                }
                j++;
            }
            i++;
        }
        
    }
}

File: 0283-move-zeroes/0283-move-zeroes.py
class Solution(object):
    def moveZeroes(self, arr: list[int]) -> None:
        i = 0
        j = 0
        while j < len(arr):
            if arr[j] != 0:
                if j != i:
                    arr[i], arr[j] = arr[j], arr[i]
                i += 1
            j += 1
